File: 1431.kids-with-the-greatest-number-of-candies.cpp
/*
 * @lc app=leetcode id=1431 lang=cpp
 *
 * [1431] Kids With the Greatest Number of Candies
 */
#include <vector>
using namespace std;
// @lc code=start
class Solution
{
public:
    vector<bool> kidsWithCandies(vector<int> &candies, int extraCandies)
    {
        int max_val = *max_element(candies.begin(), candies.end());
        vector<bool> res;
        res.reserve(candies.size());
        for (auto n : candies)
        {
            if (n + extraCandies >= max_val)
            {
                res.push_back(true);
            }
            else
            {
                res.push_back(false);
            }
        }
        return res;
    }
};
// @lc code=end

File: 1431.kids-with-the-greatest-number-of-candies.java
import java.lang.reflect.Array;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

/*
 * @lc app=leetcode id=1431 lang=java
 *
 * [1431] Kids With the Greatest Number of Candies
 */

// @lc code=start
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        List<Boolean> res = new ArrayList<>(candies.length);
        int max_val = Arrays.stream(candies).max().getAsInt();
        for (int n : candies) {
            if (n + extraCandies >= max_val) {
                res.add(true);
            } else {
                res.add(false);
            }
        }
        return res;
    }
}
// @lc code=end
